### PR TITLE
Expose artifact to a command

### PR DESCRIPTION
**Pull request by jadeallenx**

### CODE REVIEW

**Comment by Vagabond** [comment]
It'd be great to have the cli take a filename instead of dumping a big binary to stdout.

This code was very useful over the weekend, so it'd be nice to get in there to debug further group splits.

File: src/miner_consensus_mgr.erl [hunk 1]
```diff
@@ -18,6 +18,7 @@
          sign_artifact/2,
 
          %% info
+         get_artifact/0,
```
Comment (Vagabond) [inline src/miner_consensus_mgr.erl:21]: remnant?

File: src/cli/miner_cli_hbbft.erl [hunk 2]
```diff
@@ -80,6 +83,29 @@ hbbft_status(["hbbft", "status"], [], []) ->
...
+     ["hbbft artifact <filename>\n\n",
+      "  Write DKG artifact for the current consensus group to filename\n\n"
+     ]
+    ].
+
+hbbft_artifact(["hbbft", "artifact", Filename], [], []) ->
+    Response = case miner:hbbft_artifact() of
+                   not_running -> not_in_consensus;
+                   {error, timeout} -> timeout;
+                   {ok, R} -> file:write_file(Filename, R)
```
Comment (Vagabond) [inline src/cli/miner_cli_hbbft.erl:102]: artifact can be undefined here, which doesn't make sense to write to a file.

File: src/cli/miner_cli_hbbft.erl [hunk 3]
```diff
@@ -80,6 +83,29 @@ hbbft_status(["hbbft", "status"], [], []) ->
 hbbft_status([], [], []) ->
     usage.
 
+hbbft_artifact_cmd() ->
+    [
+     [["hbbft", "artifact", Filename], [], [], fun hbbft_artifact/3]
```
Comment (Vagabond) [inline src/cli/miner_cli_hbbft.erl:88]: `src/cli/miner_cli_hbbft.erl:88:29: variable 'Filename' is unbound`

File: src/cli/miner_cli_hbbft.erl [hunk 4]
```diff
@@ -80,6 +83,30 @@ hbbft_status(["hbbft", "status"], [], []) ->
...
 
+hbbft_artifact_cmd() ->
+    [
+     [["hbbft", "artifact", '*'], [], [], fun hbbft_artifact/3]
+    ].
+
+hbbft_artifact_usage() ->
+    [["hbbft", "artifact"],
+     ["hbbft artifact <filename>\n\n",
+      "  Write DKG artifact for the current consensus group to filename\n\n"
```
Comment (Vagabond) [inline src/cli/miner_cli_hbbft.erl:94]:
```suggestion
      "  Write block artifact for the current consensus group to filename\n\n"
```

File: src/miner.erl [hunk 5]
```diff
@@ -244,6 +245,21 @@ hbbft_status() ->
...
+-spec hbbft_artifact() -> {ok, undefined} | {ok, binary()} | not_running | {error, timeout}.
+hbbft_artifact() ->
+   case gen_server:call(?MODULE, consensus_group, 60000) of
+        undefined -> not_running;
+        Pid ->
+            Ref = make_ref(),
+            ok = libp2p_group_relcast:handle_input(Pid, {get_artifact, Ref, self()}),
+            receive
+                {Ref, Result} ->
+                    Result
```
Comment (Vagabond) [inline src/miner.erl:257]: should be weapped in ok tuple here, or in `handle_command({get_artifact`

Comment (jadeallenx) [reply src/miner.erl]: Fixed

**Review by Vagabond — APPROVED** [review]

**Pull request merged**